### PR TITLE
improve file filtering

### PR DESCRIPTION
**Pull request by DrPyser**
- **fix(umount): clean lsyncd process & pidfiles**
- **fix(logging): prefix logger name in verbose output**
- **fix(mount): fix sync error handling**
- **fix(mount): treat lsyncd subprocess failure as mount failure**
- **tests: tox config & coverage for mount**
- **gitignore: support gitignore rules**
- **config: support exclude in main and per-project config**
- **gitignore: emit deny-prefixed rules for filter-rule syntax**
- **mount: derive exclude rules from gitignore and config**
- **mount: always restart lsyncd on mount for fresh config**
- **main: keep filter file alive alongside its lsyncd config**

---

> [!NOTE]
> **Medium Risk**
> Changes core mount/sync behavior (always restart, stricter errors, new filter semantics) so wrong excludes or stop/kill edge cases could affect what reaches the remote dev box; risk is mitigated by tests and documented remount for `.gitignore` updates.
> 
> **Overview**
> **Mount sync** now builds rsync filter rules from built-in defaults (`.git`, `.tox`, `node_modules`), global and per-project `exclude` in config, and each repo’s `.gitignore` / `.git/info/exclude` (via new `gitignore_exclude_rules`). Configured denies are applied before gitignore rules so `!` exceptions cannot override them. **lsyncd** uses a `filterFrom` file; **rsync-only** mode passes one `--filter=` per rule.
> 
> **Remounting** always stops and restarts sync so exclude rules stay current (including when already mounted or in rsync-only mode). **Sync failures** raise `SyncError` instead of logging and continuing; mount state is not updated on failure. **Umount / stop** kills lsyncd from the stored pidfile, removes the pidfile, and clears state even when `rsync_only` toggled between mount and umount.
> 
> **Cache cleanup** treats `.filter` files as tied to their lsyncd config pidfile, prunes stale pidfiles when the process is not live (`is_lsyncd_pid`), and **verbose CLI** logs use `%(name)s:` prefixes. Docs and `config.yml.sample` describe `exclude`; **tox** runs pytest with coverage and adds broad mount/config/gitignore tests.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ed4b68384632eafe9e246da9a273e9bf426c5b48. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: wazo_sdk/gitignore.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        os.path.join(repo_path, '.git', 'info', 'exclude')
+    ) + _read_ignore_file(os.path.join(repo_path, '.gitignore'))
+
+    negations = [
+        f'+ {line[1:]}' for line in lines if line.startswith('!') and line != '!'
+    ]
+    plain = [
+        f'- {line}' for line in lines if not (line.startswith('!') and line != '!')
+    ]
+    return negations + plain
```
[the remark this answers is not in the document]
Comment (DrPyser) [reply wazo_sdk/gitignore.py:38]: I believe the implementation is a simplification of the expected pattern of negative patterns in gitignore: allowlist some patterns, then ignore everything else.

 should review this and at the very list make the conversion rules more explicit.

File: wazo_sdk/mount.py [hunk 2]
```diff
@@ -94,24 +114,22 @@ def _is_sync_running(self, mount: MountData) -> bool:
...
+        self, local_repo_name: str, repo_config: ProjectConfigData
+    ) -> list[str]:
+        local_path = os.path.join(self._local_dir, local_repo_name)
+        denies = dict.fromkeys(
+            [*self._config.exclude, *(repo_config.get('exclude') or [])]
+        )
+        return [
+            *gitignore_exclude_rules(local_path),
+            *(f'- {pattern}' for pattern in denies),
+        ]
```
[the remark this answers is not in the document]
Comment (DrPyser) [reply wazo_sdk/mount.py:132]: probably should prioritize default / hardcoded patterns indeed.
Then explicitly configured patterns in wdk config should take priority over gitignore patterns.
